macOS/UnitTests/Favicons/Mocks/CapturingFaviconImageCache.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  CapturingFaviconImageCache.swift
+//
+//  Copyright © 2025 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Common
+@testable import DuckDuckGo_Privacy_Browser
+
+@MainActor
+final class CapturingFaviconImageCache: FaviconImageCaching {
```

Comment:
Would it be better if the class name contains "Mock" for better clarity?

Reply:
I’m personally fine with both, but I noticed the distinction between “Capturing” and “Mock” classes in our codebase where mocks just provide dummy implementation while capturing classes allow for customizing function implementations and count function calls.

I could change it, but it won’t help much because there’s a lot of Capturing* classes in the codebase already :D so perhaps we would benefit from some proposal unifying the naming.